config/configcompression/compressiontype.go

```diff
@@ -3,40 +3,65 @@
 
 package configcompression // import "go.opentelemetry.io/collector/config/configcompression"
 
-import "fmt"
+import (
+	"compress/zlib"
+	"fmt"
+)
 
 // Type represents a compression method
 type Type string
 
+type Level int
+
+type CompressionParams struct {
+	Level Level `mapstructure:"level"`
+}
+
 const (
-	TypeGzip    Type = "gzip"
-	TypeZlib    Type = "zlib"
-	TypeDeflate Type = "deflate"
-	TypeSnappy  Type = "snappy"
-	TypeZstd    Type = "zstd"
-	TypeLz4     Type = "lz4"
-	typeNone    Type = "none"
-	typeEmpty   Type = ""
+	TypeGzip                Type = "gzip"
+	TypeZlib                Type = "zlib"
+	TypeDeflate             Type = "deflate"
+	TypeSnappy              Type = "snappy"
+	TypeZstd                Type = "zstd"
+	TypeLz4                 Type = "lz4"
+	typeNone                Type = "none"
+	typeEmpty               Type = ""
+	DefaultCompressionLevel      = zlib.DefaultCompression
 )
 
 // IsCompressed returns false if CompressionType is nil, none, or empty.
 // Otherwise, returns true.
-func (ct *Type) IsCompressed() bool {
-	return *ct != typeEmpty && *ct != typeNone
+func (t Type) IsCompressed() bool {
+	return t != typeEmpty && t != typeNone
 }
 
-func (ct *Type) UnmarshalText(in []byte) error {
-	typ := Type(in)
-	if typ == TypeGzip ||
-		typ == TypeZlib ||
-		typ == TypeDeflate ||
-		typ == TypeSnappy ||
-		typ == TypeZstd ||
-		typ == TypeLz4 ||
-		typ == typeNone ||
-		typ == typeEmpty {
-		*ct = typ
+func (t Type) Validate() error {
```

Comment:
Why `UnmarshalText` replaced with `Validate`? This is a breaking change. I don't think it's necessary.